### PR TITLE
Add search term in title tag and keywords on search pages

### PR DESCRIPTION
**Pull request by brunojdo**
#### What is the purpose of this pull request?
As the title says. 

#### Screenshots or example usage
https://store1--storecomponents.myvtex.com/shirt?map=ft
#### Types of changes

* [x] Bug fix (a non-breaking change which fixes an issue)
* [ ] New feature (a non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/ProductSearchContextProvider.js [hunk 1]
```diff
@@ -179,7 +184,7 @@ class ProductSearchContextProvider extends Component {
       to,
       withFacets: includeFacets(mapField, queryField),
     }
-
+    console.log("params", params)
```
Comment (klzns) [inline react/ProductSearchContextProvider.js]: 🔫

Comment (lbebber) [inline react/ProductSearchContextProvider.js]: 👀

Comment (klzns) [reply react/ProductSearchContextProvider.js]: (btw, you can do `console.log({ params })`)

File: react/ProductSearchContextProvider.js [hunk 2]
```diff
@@ -212,6 +217,7 @@ class ProductSearchContextProvider extends Component {
                 {metaTagDescription && (
                   <meta name="description" content={metaTagDescription} />
                 )}
+                {params.term && <meta name="keywords" content={`${params.term} ${metaTagKeywords}`} />}
```
Comment (klzns) [inline react/ProductSearchContextProvider.js]: keywords are comma separated

File: react/ProductSearchContextProvider.js [hunk 3]
```diff
@@ -202,10 +211,13 @@ class ProductSearchContextProvider extends Component {
...
             >
               <Helmet>
-                {titleTag && <title>{titleTag}</title>}
+                {titleTag
+                  ? <title>{titleTag}</title>
+                  : params.term && <title>{`${capitalize(params.term)} - ${storeTitle}`}</title>}
                 {metaTagDescription && (
                   <meta name="description" content={metaTagDescription} />
                 )}
+                {params.term && <meta name="keywords" content={`${params.term} ${metaTagKeywords}`} />}
```
Comment (lbebber) [inline react/ProductSearchContextProvider.js]: ⛏ 
```suggestion
                {params.term && <meta name="keywords" content={`${params.term},${metaTagKeywords}`} />}
```

File: react/ProductSearchContextProvider.js [hunk 4]
```diff
@@ -52,6 +54,9 @@ class ProductSearchContextProvider extends Component {
   static defaultProps = {
     maxItemsPerPage: DEFAULT_MAX_ITEMS_PER_PAGE,
   }
+  static contextTypes = {
```
Comment (luciannojunior) [inline react/ProductSearchContextProvider.js]: Legacy Context API should be avoided

File: react/ProductSearchContextProvider.js [hunk 5]
```diff
@@ -147,11 +152,16 @@ class ProductSearchContextProvider extends Component {
       runtime: { page: runtimePage },
     } = this.props
 
+    const settings = this.context.getSettings(APP_LOCATOR) || {}
```
Comment (luciannojunior) [inline react/ProductSearchContextProvider.js]: `runtime`  has this `getSettings` function, you should use it instead.

File: react/ProductSearchContextProvider.js [hunk 6]
```diff
@@ -202,10 +211,13 @@ class ProductSearchContextProvider extends Component {
               loading={loading}
             >
               <Helmet>
-                {titleTag && <title>{titleTag}</title>}
+                {titleTag
+                  ? <title>{titleTag}</title>
```
Comment (luciannojunior) [inline react/ProductSearchContextProvider.js]: `titleTag` wasn't renamed? Where is it coming from? And also, just use one `title` tag and make the comparison inside

Comment (brunojdo) [reply react/ProductSearchContextProvider.js]: This came from the query, the other is from settings

File: react/ProductSearchContextProvider.js (hunk 4 above)
Comment (klzns) [reply react/ProductSearchContextProvider.js]: Totally!

File: react/ProductSearchContextProvider.js (hunk 6 above)
Comment (luciannojunior) [reply react/ProductSearchContextProvider.js]: Oh, I see now... Confusing the reuse

**Pull request merged**